### ISSUE TITLE
Test for devtools seem to fail inderterministically

### ISSUE THREAD

**Issue by lzrski** [segmentio/nightmare#572]
Hello, I'm playing with Nightmare and mostly love the experience. Thanks for this tool.

It seems like electron instance is sometimes (but not always) crushing when openDevTools option is set to `true`. I was not able to find a pattern. Here is a debug log for failing case:

```
 $ DEBUG=nightmare* npm test

> nightmare@2.3.2 test /Users/tadeuszlazurski/Projects/nightmare
> make test



  Nightmare
    devtools
  nightmare queueing child action addition for "checkDevTools" +0ms
  nightmare queueing action "goto" for http://localhost:7500/simple +5ms
  nightmare queueing action "wait" +1ms
  nightmare queueing action "checkDevTools" +0ms
  nightmare running +0ms
  nightmare:log adding action for checkDevTools +289ms
  nightmare:log did-start-loading +57ms [{"sender":{"session":{"cookies":{},"webRequest":{}},"hostWebContents":null,"devToolsWebContents":null,"debugger":{},"_maxListeners":0,"_events":{"page-title-updated":[null,null],"crashed":[null,null]},"_eventsCount":28,"browserWindowOptions":{"show":true,"openDevTools":true,"alwaysOnTop":true,"webPreferences":{"preload":"/Users/tadeuszlazurski/Projects/nightmare/lib/preload.js","nodeIntegration":false}}}}]
  nightmare:log did-get-redirect-request +38ms [{"sender":{"session":{"cookies":{},"webRequest":{}},"hostWebContents":null,"devToolsWebContents":null,"debugger":{},"_maxListeners":0,"_events":{"page-title-updated":[null,null],"crashed":[null,null],"did-finish-load":[null,null]},"_eventsCount":27,"browserWindowOptions":{"show":true,"openDevTools":true,"alwaysOnTop":true,"webPreferences":{"preload":"/Users/tadeuszlazurski/Projects/nightmare/lib/preload.js","nodeIntegration":false}}}},"http://localhost:7500/simple","http://localhost:7500/simple/",true,303,"GET","",{"connection":["keep-alive"],"content-length":["47"],"content-type":["text/html; charset=UTF-8"],"date":["Fri, 15 Apr 2016 08:34:10 GMT"],"location":["/simple/"],"x-content-type-options":["nosniff"],"x-powered-by":["Express"]}]
  nightmare:log did-get-response-details +13ms [{"sender":{"session":{"cookies":{},"webRequest":{}},"hostWebContents":null,"devToolsWebContents":null,"debugger":{},"_maxListeners":0,"_events":{"page-title-updated":[null,null],"crashed":[null,null],"did-finish-load":[null,null]},"_eventsCount":27,"browserWindowOptions":{"show":true,"openDevTools":true,"alwaysOnTop":true,"webPreferences":{"preload":"/Users/tadeuszlazurski/Projects/nightmare/lib/preload.js","nodeIntegration":false}}}},false,"http://localhost:7500/simple/","http://localhost:7500/simple",200,"GET","",{"accept-ranges":["bytes"],"cache-control":["public, max-age=0"],"connection":["keep-alive"],"content-length":["121"],"content-type":["text/html; charset=UTF-8"],"date":["Fri, 15 Apr 2016 08:34:10 GMT"],"etag":["W/\"79-15415804728\""],"last-modified":["Thu, 14 Apr 2016 16:00:09 GMT"],"x-powered-by":["Express"]}]
  nightmare:log dom-ready +1ms [{"sender":{"session":{"cookies":{},"webRequest":{}},"hostWebContents":null,"devToolsWebContents":null,"debugger":{},"_maxListeners":0,"_events":{"page-title-updated":[null,null],"crashed":[null,null],"did-finish-load":[null,null]},"_eventsCount":27,"browserWindowOptions":{"show":true,"openDevTools":true,"alwaysOnTop":true,"webPreferences":{"preload":"/Users/tadeuszlazurski/Projects/nightmare/lib/preload.js","nodeIntegration":false}}}}]
  nightmare:log did-frame-finish-load +15ms [{"sender":{"session":{"cookies":{},"webRequest":{}},"hostWebContents":null,"devToolsWebContents":null,"debugger":{},"_maxListeners":0,"_events":{"page-title-updated":[null,null],"crashed":[null,null],"did-finish-load":[null,null]},"_eventsCount":27,"browserWindowOptions":{"show":true,"openDevTools":true,"alwaysOnTop":true,"webPreferences":{"preload":"/Users/tadeuszlazurski/Projects/nightmare/lib/preload.js","nodeIntegration":false}}}},true]
  nightmare:log did-finish-load +0ms [{"sender":{"session":{"cookies":{},"webRequest":{}},"hostWebContents":null,"devToolsWebContents":null,"debugger":{},"_maxListeners":0,"_events":{"page-title-updated":[null,null],"crashed":[null,null],"did-finish-load":[null,null]},"_eventsCount":27,"browserWindowOptions":{"show":true,"openDevTools":true,"alwaysOnTop":true,"webPreferences":{"preload":"/Users/tadeuszlazurski/Projects/nightmare/lib/preload.js","nodeIntegration":false}}}}]
  nightmare:log page-favicon-updated +0ms [{"sender":{"session":{"cookies":{},"webRequest":{}},"hostWebContents":null,"devToolsWebContents":null,"debugger":{},"_maxListeners":0,"_events":{"page-title-updated":[null,null],"crashed":[null,null],"did-finish-load":[null]},"_eventsCount":27,"browserWindowOptions":{"show":true,"openDevTools":true,"alwaysOnTop":true,"webPreferences":{"preload":"/Users/tadeuszlazurski/Projects/nightmare/lib/preload.js","nodeIntegration":false}}}},["http://localhost:7500/favicon.ico"]]
  nightmare:log did-stop-loading +0ms [{"sender":{"session":{"cookies":{},"webRequest":{}},"hostWebContents":null,"devToolsWebContents":null,"debugger":{},"_maxListeners":0,"_events":{"page-title-updated":[null,null],"crashed":[null,null],"did-finish-load":[null]},"_eventsCount":27,"browserWindowOptions":{"show":true,"openDevTools":true,"alwaysOnTop":true,"webPreferences":{"preload":"/Users/tadeuszlazurski/Projects/nightmare/lib/preload.js","nodeIntegration":false}}}}]
  nightmare:actions .wait() for 2000ms +1ms
  nightmare electron child process exited with code null: undefined +396ms
      1) should open devtools
  nightmare running +9s
      2) "after each" hook for "should open devtools"


  0 passing (20s)
  2 failing

  1) Nightmare devtools should open devtools:
     Error: timeout of 10000ms exceeded. Ensure the done() callback is being called in this test.


  2) Nightmare devtools "after each" hook for "should open devtools":
     Error: timeout of 10000ms exceeded. Ensure the done() callback is being called in this test.




make: *** [test] Error 2
npm ERR! Test failed.  See above for more details.
```

I've been also seeing this behaviour with real use (i.e. outside the tests). Might be related to #563.

**Comment by rosshinkley** [comment]
@lzrski I have also been having problems with Nightmare + devtools options spontaneously flaking.  It looks like a segfault internal to Electron, just like #563.  I'll have to carve out some time to see what the details are.

**Comment by Mr0grog** [comment]
Ha! I spent all day isolating this in Election (https://github.com/electron/electron/issues/5180) and never even saw this issue :P

There’s a patch for Nightmare: #576. It just needs review.

**Comment by rosshinkley** [comment]
@Mr0grog Thank you for putting in the legwork.  Hopefully that electron issue will be resolved soon.

**Comment by rosshinkley** [comment]
Should be fixed with #588.  Closing.
